### PR TITLE
add nextjs wagmi 0.12.x example

### PR DESCRIPTION
**Pull request by glitch-txs**
Addressing https://github.com/orgs/WalletConnect/discussions/2649?notification_referrer_id=NT_kwDOA_2SuLM2Nzc2MTAxNzMxOjY2OTQ5ODE2

Keeping wagmi 0.12.x example. I wonder if this is gonna be stable for WalletConnect as well?

### CODE REVIEW

File: nextjs-wagmi-v0.12.x/src/pages/_app.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  bsc,
+  optimism,
+  gnosis,
+  fantom,
+];
+
+const { provider } = configureChains(chains, [w3mProvider({ projectId })]);
+const wagmiClient = createClient({
+  autoConnect: true,
+  connectors: w3mConnectors({ version: 1, chains, projectId }),
```
Comment (MidnightLightning) [inline nextjs-wagmi-v0.12.x/src/pages/_app.jsx:42]: This still targets WalletConnect v1 with this parameter, I believe?

Comment (glitch-txs) [reply nextjs-wagmi-v0.12.x/src/pages/_app.jsx]: correct, should be 2

**Review by MidnightLightning — COMMENTED** [review]
Confirming this works for me in my setup, even going up to

```
"@web3modal/ethereum": "2.3.0",
"@web3modal/react": "2.3.0",
"wagmi": "^0.12.16"
```

**Comment by MidnightLightning** [comment]
Dropping a ping to @0xAsimetriq and @finessevanes; is there more that's needed to review this merge request? We're now past the WalletConnect v2 upgrade deadline, but making this example more visible to those in this situation would be helpful.

**Comment by FilipMasar** [comment]
Thanks for this example. It really helped a lot. I encountered the same issue when I wanted to continue using wagmi. To make this example work I had to include `"@walletconnect/modal": "^2.6.1",` in dependencies

**Pull request closed without merging**